### PR TITLE
Support repositories loaded from treeinfo file

### PR DESCRIPTION
**Pull request by jkonecny12**
Load base repo path and addition repositories from the base repository root.

These additional repositories loaded from the base repo path can't be modified by a user and can't be removed. They will be disabled first time (for each source spoke visit) when the base repository changed. This should give a user hint that these repositories are related to the base repository.

  *Resolves: rhbz#1610806*

This PR partially deprecate the PR https://github.com/rhinstaller/anaconda/pull/1540 .

### CODE REVIEW

File: pyanaconda/payload/dnfpayload.py [hunk 1]
```diff
@@ -1104,8 +1105,11 @@ def reset(self):
     def updateBaseRepo(self, fallback=True, checkmount=True):
         log.info('configuring base repo')
         self.reset()
-        url, mirrorlist, metalink = self._setupInstallDevice(self.storage,
-                                                             checkmount)
+        repo_root_url, mirrorlist, metalink = self._setupInstallDevice(self.storage, checkmount)
```
Comment (poncovka) [inline pyanaconda/payload/dnfpayload.py]: What is wrong with the name `url`?

File: pyanaconda/payload/__init__.py [hunk 2]
```diff
@@ -995,7 +1010,8 @@ def __init__(self, data):
 class PackagePayload(Payload):
     """A PackagePayload installs a set of packages onto the target system."""
 
-    DEFAULT_REPOS = [productName.split('-')[0].lower(), "fedora-modular-server", "rawhide"]  # pylint: disable=no-member
+    DEFAULT_REPOS = [productName.split('-')[0].lower(), "fedora-modular-server", "rawhide",
+                     "BaseOS"]  # pylint: disable=no-member
```
Comment (poncovka) [inline pyanaconda/payload/__init__.py]: Please, split this list to multiple lines.

File: pyanaconda/payload/dnfpayload.py [hunk 3]
```diff
@@ -1207,6 +1216,47 @@ def updateBaseRepo(self, fallback=True, checkmount=True):
...
                     self._fetch_md(repo)
 
+    def _getBaseRepoUrl(self, url):
+        """Try to find base repository from the treeinfo file."""
+        if self._treeinfo:
+            variants = self._treeinfo.variants
+
+            for variant in variants:
+                if variant in self.DEFAULT_REPOS:
+                    variant_obj = variants[variant]
```
Comment (poncovka) [inline pyanaconda/payload/dnfpayload.py]: Please, iterate over keys and items to avoid this.

File: pyanaconda/payload/dnfpayload.py [hunk 4]
```diff
@@ -1207,6 +1216,47 @@ def updateBaseRepo(self, fallback=True, checkmount=True):
...
+                    log.info("Found base repository in treeinfo file")
+                    if variant_obj.paths.repository == ".":
+                        log.debug("Treeinfo points base repository to installation tree root")
+                    else:
+                        url = url + "/" + variant_obj.paths.repository
+                        log.debug("Treeinfo points base repository to %s", url)
```
Comment (poncovka) [inline pyanaconda/payload/dnfpayload.py]: I don't understand the messages.

File: pyanaconda/payload/dnfpayload.py [hunk 5]
```diff
@@ -1207,6 +1216,47 @@ def updateBaseRepo(self, fallback=True, checkmount=True):
...
+                    return url
+
+        log.debug("No base repository found in treeinfo file. Using installation tree root")
```
Comment (poncovka) [inline pyanaconda/payload/dnfpayload.py]: Missing dot.

File: pyanaconda/payload/dnfpayload.py [hunk 6]
```diff
@@ -1207,6 +1216,47 @@ def updateBaseRepo(self, fallback=True, checkmount=True):
...
+        return url
+
+    def _add_treeinfo_repositories(self, repo_root_url, base_repo_url):
```
Comment (poncovka) [inline pyanaconda/payload/dnfpayload.py]: The names of variables are really confusing.

File: pyanaconda/payload/dnfpayload.py (hunk 3 above)
Comment (jkonecny12) [reply pyanaconda/payload/dnfpayload.py]: Yeah I wanted to but GTK doesn't allow that :(

File: pyanaconda/payload/dnfpayload.py (hunk 4 above)
Comment (jkonecny12) [reply pyanaconda/payload/dnfpayload.py]: As I explained above, you can point to a repository or an repository root.
In this case the treeinfo path of the repository is `.` -- in other words it is repository root.

File: pyanaconda/payload/dnfpayload.py (hunk 6 above)
Comment (jkonecny12) [reply pyanaconda/payload/dnfpayload.py]: We can change this, any ideas?

File: pyanaconda/payload/dnfpayload.py (hunk 3 above)
Comment (poncovka) [reply pyanaconda/payload/dnfpayload.py]: Ok :-)

**Comment by jkonecny12** [comment]
**UPDATED:**
* Fix issues pointed by @poncovka 
* Rename base_repo_root to installation_tree_root
* Add commit to disable additional treeinfo based repositories just once when the base repo changes

**Comment by jkonecny12** [comment]
**WIP -- DO NOT MERGE**

**Pull request merged**